test/createDB.spec.js

```diff
@@ -1,106 +1,34 @@
-                "database" : "nodeDB",
-                "options" : { "index.type" : "disk" },
-                "files": []
-            };
-
-            conn.createDB(options, function (data, response) {
-                expect(response.statusCode).to.be(409);
-
-                done();
-            });
-        });
-
-        it ("should create a new empty DB, returning 201", function (done) {
-
-            var options = {
-                "database" : "nodeDB2",
-                "options" : { "search.enabled" : true },
-                "files" : []
-            };
-
-            // filesArr.push({
-            //  "name" : "api_tests.nt"
-            // });
-            
-            conn.createDB(options, function (data, response) {
-                expect(response.statusCode).to.be(201);
-                // console.log(data);
-
-                if (response.statusCode == 201) {
-                    // clean created DB after we know it was created
-                    conn.dropDB({ database: "nodeDB2" }, function (data2, response2) {
-                        expect(response2.statusCode).to.be(200);
-
-                        done();
-                    });
-                }
-            });
-
-        });
-
-        // it ("should create a new DB with data, returning 201", function (done) {
-        //  var options = {
-        //      "search.enabled" : true
-        //  };
-
-        //  var filesArr = [];
-        //  filesArr.push({
-        //      "name" : "api_tests.nt"
-        //  });
-            
-        //  conn.createDB("newNodeDB", options, filesArr, function (data, response) {
-        //      expect(response.statusCode).toBe(201);
-        //      console.log(data);
-
-        //      if (response.statusCode === 201) {
-        //          conn.dropDB("newNodeDB", function (data2, response2) {
-        //              expect(response2.statusCode).toBe(200);
-
-        //              done();
-        //          });
-        //      }
-        //      else {
-        //          // console.log(data);
-        //          done();
-        //      }
-        //  }, "data/api_tests.nt");
-
-        // });
-
+const Stardog = require('../lib');
+const {
+  seedDatabase,
+  dropDatabase,
+  generateDatabaseName,
+} = require('./setup-database');
+
+describe('createDB()', () => {
+  const database = generateDatabaseName();
+  var conn;
+
+  beforeAll(seedDatabase(database));
+  afterAll(dropDatabase(database));
+
+  beforeEach(() => {
+    conn = new Stardog.Connection();
+    conn.setEndpoint('http://localhost:5820/');
+    conn.setCredentials('admin', 'admin');
+  });
+
+  it('should not be able to create a new db with the name of an existing DB', done => {
+    var options = {
+      database,
+      options: { 'index.type': 'disk' },
+      files: [],
+    };
+
+    conn.createDB(options, (data, response) => {
+      expect(response.statusCode).toEqual(409);
+
+      done();
     });
-}));
\ No newline at end of file
+  });
```

Comment:
For now yes because we do those things over and over again in each test. I test to make sure the status code is 201.